### ISSUE TITLE
Importing/overwriting a channel but not saving it causes the client to be out of sync with the server

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#2878]
* Create a new channel, and save it.
* Export the channel.
* Edit the channel again, change something besides the name (like the source connector type), and save it.
* Import the channel you exported before, and choose to overwrite the existing channel.
* Return to the Channels view without saving.

If you edit the channel again, you'll see that it has reverted to the first version (the one you exported), even though you never saved it. However if you deploy the channel, it will deploy the **new** version (e.g. the one with the changed source connector type). So now the client and server are out of sync and will be until the channel is saved again.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-2953
Reporter: narupley
Created: 2013-09-26T07:45:53.000-0700

**Comment by rbeckman-nextgen** [comment]
Fixed with revisions 6977 and 6978. We're now always updating the channel beforehand if the user is importing/overwriting, dragging over multiple channels, or if the channel is invalid. When the user imports and overwrites a single channel, the Edit Channel view is still shown, but the save button is automatically disabled because we've already saved the channel for them.

Imported Comment. Original Details:
Author: narupley
Created: 2013-10-15T16:22:29.000-0700
